src/test/java/bio/terra/service/snapshotbuilder/query/QueryTest.java

```diff
@@ -122,4 +129,82 @@ void renderComplexSQL() {
                 "(c.condition_concept_id = 316139 OR c0.ancestor_concept_id = 316139 OR c.condition_concept_id = 4311280 OR c0.ancestor_concept_id = 4311280)"),
             containsString("AND p.year_of_birth < 1983")));
   }
+
+  @Test
+  void renderSQLWithDatasetModelGCP() {
+    DatasetModel dataset =
+        new DatasetModel().name("name").dataProject("project").cloudPlatform(CloudPlatform.GCP);
+    Query query = buildGetConceptsQuery(100, generateTableNameGCP(dataset));
+    String sql = query.renderSQL();
+    assertThat(
+        sql,
+        is(
+            "SELECT c.concept_name, c.concept_id FROM `project.datarepo_name.concept` AS c "
+                + "WHERE c.concept_id IN "
+                + "(SELECT c.descendant_concept_id FROM `project.datarepo_name.concept_ancestor` AS c "
+                + "WHERE c.ancestor_concept_id = 100)"));
+  }
+
+  @Test
+  void renderSQLWithDatasetModelAzure() {
+    String datasetSource = "source_dataset_data_source_0";
+    Query query = buildGetConceptsQuery(100, generateTableNameAzure(datasetSource));
+    String sql = query.renderSQL();
+    assertThat(
+        sql,
+        is(
+            "SELECT c.concept_name, c.concept_id FROM "
+                + "(SELECT * FROM OPENROWSET(BULK 'metadata/parquet/concept/*/*.parquet', DATA_SOURCE = 'source_dataset_data_source_0', FORMAT = 'parquet') AS alias951024263) "
+                + "AS c WHERE c.concept_id IN "
+                + "(SELECT c.descendant_concept_id FROM "
+                + "(SELECT * FROM OPENROWSET(BULK 'metadata/parquet/concept_ancestor/*/*.parquet', DATA_SOURCE = 'source_dataset_data_source_0', FORMAT = 'parquet') AS alias625571305) "
+                + "AS c WHERE c.ancestor_concept_id = 100)"));
```

Comment:
I'm working on testing this as soon as I get the Azure connected test working!

Additionally, I think it will be good to take a look at this PR once the GCP version is merged since it looks like this includes GCP changes as well.

Reply:
Confirmed that this query works!
* [the test](https://github.com/DataBiosphere/jade-data-repo/pull/1574/files#diff-d73b047089637f4497d5947376822a2fd247755d50c84e4b2566e8ac1ae8abffR383)
* [the query](https://github.com/DataBiosphere/jade-data-repo/pull/1574/files#diff-66aef91d0499d11e775c637c19bc02cb5fdda1e879e414286a5f0354d9693f2eR83)